app/layout/account-bar.jsx

```diff
@@ -17,55 +16,57 @@ const DOWN = 40;
 
 counterpart.registerTranslations('en', {
   accountMenu: {
+    messages: 'Messages',
     profile: 'Profile',
     home: 'Home',
     settings: 'Settings',
     signOut: 'Sign Out',
     collections: 'Collections',
-    favorites: 'Favorites',
-  },
+    favorites: 'Favorites'
+  }
 });
 
-const AccountBar = React.createClass({
-  contextTypes: {
-    user: React.PropTypes.object,
-    router: routerShape,
-    geordi: React.PropTypes.object,
-  },
-
-  getInitialState() {
-    return {
-      unread: false,
+export default class AccountBar extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      messageCount: 0,
+      unread: false
     };
-  },
+
+    this.handleAccountMenuOpen = this.handleAccountMenuOpen.bind(this);
+    this.navigateMenu = this.navigateMenu.bind(this);
+    this.handleSignOutClick = this.handleSignOutClick.bind(this);
+  }
 
   componentDidMount() {
     addEventListener('locationchange', this.lookUpUnread);
     this.lookUpUnread();
```

Comment:
For this listener to work, `this.lookUpUnread` should be bound to the component's `this` context in the constructor.